Remove Quirk simulatedMouseEventTypeForTarget for maps.google.com

karlcow opened the pull request:
#### e27a1b42d29ec9f4d8fbee13761f91246cf7a7e7
<pre>
Remove Quirk simulatedMouseEventTypeForTarget for maps.google.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=279250">https://bugs.webkit.org/show_bug.cgi?id=279250</a>
<a href="https://rdar.apple.com/135400991">rdar://135400991</a>

Reviewed by Brent Fulgham.

This removes the quirk which was dead code.
Google Maps has been contacted and fixed their code in a way that
the yellow pegman icon is draggable on Google Maps.

Their is a counterpart for this removal on Safari source code.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::simulatedMouseEventTypeForTarget const): Deleted.
* Source/WebCore/page/Quirks.h:

Canonical link: <a href="https://commits.webkit.org/283859@main">https://commits.webkit.org/283859@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/adf67d5dd19369fb89c18ccb36db1bcde98191c3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67440 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46819 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20072 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71488 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18577 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69558 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54617 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18368 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54056 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12444 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70507 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42986 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58333 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34518 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39659 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15736 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16935 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61623 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16077 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73187 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11399 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15399 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61496 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11434 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58401 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61555 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14991 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9320 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2935 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42624 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43701 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44887 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43442 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
This change is fine, but we cannot land it until the relevant Safari code has updated to expect this method to have been removed.

Please do not land this change until that internal change is committed.

brentfulgham reviewed (changes requested):
Upon further review, I don't think this is right. See comments below.

brentfulgham commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -537,22 +537,6 @@ bool Quirks::shouldDispatchedSimulatedMouseEventsAssumeDefaultPrevented(EventTar
 }
 
-// maps.google.com https://bugs.webkit.org/show_bug.cgi?id=199904
-std::optional<Event::IsCancelable> Quirks::simulatedMouseEventTypeForTarget(EventTarget* target) const
-{
-    if (!shouldDispatchSimulatedMouseEvents(target))
-        return { };
-
-    // On Google Maps, we want to limit simulated mouse events to dragging the little man that allows entering into Street View.
-    if (isGoogleMaps()) {
-        if (RefPtr element = dynamicDowncast<Element>(target); element && element->getAttribute(HTMLNames::classAttr) == "widget-expand-button-pegman-icon"_s)
-            return Event::IsCancelable::Yes;
-        return { };
-    }
-
-    return Event::IsCancelable::Yes;
```
I think we still need to return `Event::IsCancelable::Yes` when `shouldDispatchSimulatedMouseEvents(target)` is true, so we may just need to revise this to:

```
std::optional<Event::IsCancelable> Quirks::simulatedMouseEventTypeForTarget(EventTarget* target) const)
{
    return shouldDispatchSimulatedMouseEvents(target) ? Event::IsCancelable::Yes : { };
}
```

Alternatively, we could remove this entire method from here, but we would need to make more changes to the Safari code to avoid impacting quirks used for desktop iPad sites.

brentfulgham replied:
@karlcow and I spoke in person, and decided deleting this unused code is a better approach, and will change the Safari bit to match.

brentfulgham reviewed (approved):
Let's land the internal bit first, then this can land.

karlcow committed:
Remove Quirk simulatedMouseEventTypeForTarget for maps.google.com
https://bugs.webkit.org/show_bug.cgi?id=279250
rdar://135400991

Reviewed by Brent Fulgham.

This removes the quirk which was dead code.
Google Maps has been contacted and fixed their code in a way that
the yellow pegman icon is draggable on Google Maps.

Their is a counterpart for this removal on Safari source code.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::simulatedMouseEventTypeForTarget const): Deleted.
* Source/WebCore/page/Quirks.h:

Canonical link: https://commits.webkit.org/283859@main
